syft/frameworks/torch/tensors/interpreters/promise.py

```diff
@@ -0,0 +1,200 @@
+import syft as sy
+from syft.workers.abstract import AbstractWorker
+import weakref
+
+from syft.generic.tensor import AbstractTensor
+from syft.generic.tensor import initialize_tensor
+from syft.messaging.promise import Promise
+from syft.generic.frameworks.hook import hook_args
+
+
+class PromiseTensor(AbstractTensor, Promise):
+    def __init__(
+        self,
+        shape,
+        owner=None,
+        id=None,
+        tags=None,
+        description=None,
+        tensor_id=None,
+        tensor_type=None,
+        plans=None,
+    ):
+        """Initializes a PromiseTensor
+
+        Args:
+            owner: An optional BaseWorker object to specify the worker on which
+                the tensor is located.
+            id: An optional string or integer id of the LoggingTensor.
+        """
+
+        if owner is None:
+            owner = sy.local_worker
+
+        # I did check that the two __init__ methods below only get called once, but it
+        # was exhibiting some strange behavior when I used super() for both of them.
+
+        # constructor for AbstractTensor
+        super().__init__(id=id, owner=owner, tags=tags, description=description)
+
+        # constructor for Promise
+        Promise.__init__(self, owner=owner, obj_type=tensor_type, plans=plans)
+
+        self._shape = shape
+
+        del self.child
+
+    def torch_type(self):
+        return self.obj_type
+
+    @property
+    def shape(self):
```

Comment:
Why exactly is it a property?
Does it need to be "un-changeable" ?

Reply:
I guess it's because it is a property for AbstractObject, which is a parent class of Promise

Reply:
This should absolutely be a property. What a property means in python is that it returns the value when you do not add parenthesis to it